### ISSUE TITLE
Refactor (and delete?) the `src/mixins/base-input.js::change()` method

### ISSUE THREAD

**Issue by volterra79** [g3w-suite/g3w-client#393]
### Checklist

- [X] I've searched through the [current issues](https://github.com/g3w-suite/g3w-client/issues) to make sure this feature hasn't been requested already.

### Motivation

There is a lot of confusion about input change of `state.value` fields (form's input).

It can be changed by input itself or in some other way (eg. result of `default_expression`).

### Suggested solution

Remove change (mixin base-input) method:

https://github.com/g3w-suite/g3w-client/blob/9118100fa659415a23c628e736fb94ecfd3cf0ad/src/mixins/base-input.js#L36-L44

in favor of a common `state.value` watcher (or another way).

**Comment by Raruto** [comment]
I know it's a much broader topic, but I'm also of the opinion that some classes and functions (in [here](https://github.com/g3w-suite/g3w-client)) could be moved directly into the [editing](https://github.com/g3w-suite/g3w-client-plugin-editing) plugin.

If then in the future some similar features will be necessary within other plugins (eg. editing some of the layer properties), it would be safier to re-implement that functionality within each "external" plugin (or that at least that this third party plugin should relies mostly on classes provided by the [editing](https://github.com/g3w-suite/g3w-client-plugin-editing) plugin, and not somewhere in between) .

**Comment by volterra79** [comment]
@Raruto I agree. It is just a remainder to don't forget to refactor editing in better way
